### DISCUSSION TITLE
How to upgrade directus?

### DISCUSSION THREAD

**Discussion by joe-blocher** [directus/directus#9581] [Archive]
I wanted to upgrade from version v9.0.0-rc.95 by running the commands:

**npm update
npx directus bootstrap** 

The result was:
- a blank screen when logged in as admin
- no more tables in database

I had to restore from a backup...

**Comment by iqraaswad** [comment]
use  **npx directus database migrate:latest** after **npm update**... refers to https://docs.directus.io/configuration/upgrades-migrations.

**Comment by azrikahar** [comment]
The result you mentioned is indeed very odd and shouldn't really happen. Is the db sqlite by any chance?

If the problem still persists after following the upgrade steps outlined in the docs, do feel free to open an issue so that we can track it.

**Reply by joe-blocher** [reply]
The db is mariadb. I followed the steps of upgrades-migrations:

- **npm update**

   The result was an empty screen when opened the admin in the browser. couldn't log in because of the following errors:

![image](https://user-images.githubusercontent.com/93246850/140707583-9eb6323a-98d6-4cfe-9729-a45f90c6ba1f.png)

- **npx directus database migrate:latest**

works! 
db is not empty,  the tables where not deleted.
      
I think I open an issue.

**Reply by joe-blocher** [reply]
**update:**
restarted the server - no errors any more.
Everything is fine :-)

But I will open an issue because "Add existing" when I want to add an existing file to filelist.
All other files in the list where deleted. This is not what I expected!
When I "Upload" a file it works.

**Reply by azrikahar** [reply]
Glad it is resolved!

> All other files in the list where deleted. This is not what I expected!

Were you using Docker? this isn't exactly a Directus issue, but more so happens when you migrate without backing up the files (I'd assume it's local since it shouldn't happen if you're using something like S3).

**Reply by joe-blocher** [reply]
I don't use Docker, it's a own debian-server with a few websites (vue, express, directus, php: yii, typo3 ...).
Love the combination of vue and directus. It's perfect for smaller websites.
But I was happy to have a backup of the server from last day :-)
